src/argilla/client/feedback/schemas/questions.py

```diff
@@ -153,9 +154,34 @@ def update_settings(cls, values: Dict[str, Any]) -> Dict[str, Any]:
             ]
         if isinstance(values.get("labels"), list):
             values["settings"]["options"] = [{"value": label, "text": label} for label in values.get("labels")]
-        values["settings"]["visible_options"] = values.get(
-            "visible_labels"
-        )  # `None` is a possible value, which means all labels are visible
+        visible_options = values.get("visible_labels")
+        total_options = len(values["settings"]["options"])
+        if visible_options and visible_options > total_options:
+            if total_options >= 3:
+                warnings.warn(
+                    f"`visible_labels={visible_options}` is greater than the total number"
+                    f" of labels ({total_options}), so it will be set to `{total_options}`.",
+                    UserWarning,
+                    stacklevel=1,
+                )
+                visible_options = total_options
+            else:
+                warnings.warn(
+                    f"`labels={values.get('labels')}` has less than 3 labels, so `visible_labels`"
+                    " will be set to `None`, which means that all the labels will be visible.",
+                    UserWarning,
+                    stacklevel=1,
+                )
+                visible_options = None
+        if not visible_options and total_options > 20:
```

Comment:
This will fail in one use case. I want to show all the labels and I have 25 different labels. I need to force `visible_labels=25` instead of forcing it to `None` to show all of them.

I would say to keep the same behavior for all cases. 

* If I want to show all the labels and `total_options  <= 20`, I can force by setting `visible_labels=None`
* If I want to show all the labels and `total_options > 20`, I cannot use `visible_options=None` I need to pass the exact number of labels.

I think we can differentiate if `visible_labels` is not notified from an explicit configuration to `None`. We could check if `visible_labels` is present in `values` or not. If is present and is `None` we should let the user decision to show all the labels.